test/functional/Java11andUp/src/org/openj9/test/varhandle/TestVarHandleInfo.java

```diff
@@ -0,0 +1,115 @@
+
+	protected static Logger logger = Logger.getLogger(TestVarHandleInfo.class);
+	private static final String TEST_VAR_HANDLE_INFO = "Lorg/openj9/test/varhandle/TestVarHandleInfo;"; //$NON-NLS-1$
+	private static final String INTEGER = "Ljava/lang/Integer;"; //$NON-NLS-1$
+	private static final int PUBLIC_FINAL_NATIVE = Modifier.PUBLIC + Modifier.FINAL + Modifier.NATIVE;
+	private static final int expectedModifiers = PUBLIC_FINAL_NATIVE;
+	private static final String EXPECTED_MODIFIERS_STRING = Modifier.toString(expectedModifiers);
+	private static final int expectedReferenceKind = MethodHandleInfo.REF_invokeVirtual;
+	
+	Lookup mylookup;
+	Integer instanceField; /* the target of the varhandle */
+	List<String> accessModes;
+	HashMap<String, MethodType> infos;
+
+	@Test
+	public void testAccessModes() throws NoSuchFieldException, IllegalAccessException {
+		VarHandle vh = mylookup.findVarHandle(TestVarHandleInfo.class, "instanceField", Integer.class); //$NON-NLS-1$
+		for (String methName: accessModes) {
+			logger.debug(methName);
+			VarHandle.AccessMode am = VarHandle.AccessMode.valueFromMethodName(methName);
+			MethodHandle mh = MethodHandles.varHandleExactInvoker(am,vh.accessModeType(am));
+			checkMethodHandleInfo(mylookup.revealDirect(mh), infos.get(methName), methName);
+
+			mh = MethodHandles.varHandleInvoker(am,vh.accessModeType(am));
+			checkMethodHandleInfo(mylookup.revealDirect(mh), infos.get(methName), methName);
+		}
+	}
+
+	private static void checkMethodHandleInfo(MethodHandleInfo info, MethodType expectedMethodType, String expectedMethodName) {	
+		assertEquals("Incorrect declaring class", VarHandle.class, info.getDeclaringClass()); //$NON-NLS-1$	
+
+		assertEquals("Incorrect methodType", expectedMethodType, info.getMethodType()); //$NON-NLS-1$
+		
+		int actualModifiers = info.getModifiers();
+		assertTrue("Missing modifiers: expected \""+EXPECTED_MODIFIERS_STRING  //$NON-NLS-1$
+				+ "\" (" + Integer.toHexString(expectedModifiers)  //$NON-NLS-1$
+		+ "), actual \"" + Modifier.toString(actualModifiers)  //$NON-NLS-1$
+		+ "\" ("+Integer.toHexString(actualModifiers)+")",  //$NON-NLS-1$ //$NON-NLS-2$
+		(expectedModifiers & actualModifiers) == expectedModifiers);
+		
+		assertEquals("Incorrect name ", expectedMethodName, info.getName()); //$NON-NLS-1$
+		
+		assertEquals("Incorrect reference kind, expected " //$NON-NLS-1$
+		+ MethodHandleInfo.referenceKindToString(expectedReferenceKind), 
+				expectedReferenceKind, info.getReferenceKind());
+	}
+
+	@BeforeSuite
+	public void makeInfos() {
+		infos = new HashMap<>();
+		mylookup = MethodHandles.lookup();
+		accessModes = Arrays.asList(VarHandle.AccessMode.values())
+				.stream().map(m -> m.methodName()).collect(Collectors.toList());
+		
+		accessModes.stream().filter(m -> m.contains("get")) //$NON-NLS-1$
+		.forEach(s -> infos.put(s, MethodType.fromMethodDescriptorString("("+TEST_VAR_HANDLE_INFO +")" + INTEGER, null))); //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
As the MethodType is constant for this loop, why not compute it once outside the forEach and reuse it as creating MTs is expensive?

```
MethodType getType = MethodType.fromMethodDescriptorString("("+TEST_VAR_HANDLE_INFO +")" + INTEGER, null);
accessModes.stream().filter(m -> m.contains("get")).forEach(s -> infos.put(s, getType));
```

I think the more imperative version of this code would be more readable and would occur in a single pass:
```java
for (AccessMode am : VarHandle.AccessMode.values()) {
  String m = am.methodName();
  if (m.contains("get") { 
      infos.put(m, getType); 
  } else if (m.contains("set")) {
     infos.put(m, setType);
  } else if ....
}
```
but won't insist on it being changed